knebind/solver/solver.go

```diff
@@ -89,15 +88,65 @@ func Solve(tb *opb.Testbed, topo *tpb.Topology) (*binding.Reservation, error) {
 		dstDev := id2Dev[dstParts[0]]
 		srcPort := s.dev2Ports[srcDev][srcParts[1]]
 		dstPort := s.dev2Ports[dstDev][dstParts[1]]
-		dLink := &devLink{srcDev: srcDev, srcPort: srcPort, dstDev: dstDev, dstPort: dstPort}
-		s.dev2Links[srcDev][dstDev] = append(s.dev2Links[srcDev][dstDev], dLink)
-		s.dev2Links[dstDev][srcDev] = append(s.dev2Links[dstDev][srcDev], dLink)
+		group := srcPort.GetGroup()
+		if group == "" {
+			group = dstPort.GetGroup()
+		}
+		devLinkPtr := &devLink{srcDev: srcDev, dstDev: dstDev, srcPort: srcPort, dstPort: dstPort, group: group}
```

Comment:
Rename to "dLink" for consistency with the rest of the file. Go style is typically to not include any part of the type in the variable name.